src/Persistance/RepositoryImplementations/SystemRegisterRepository.cs

```diff
@@ -58,17 +59,19 @@ public async Task<List<RegisteredSystem>> GetAllActiveSystems()
     }
 
     /// <inheritdoc/>  
-    public async Task<Guid?> CreateRegisteredSystem(RegisteredSystem toBeInserted)
+    public async Task<Guid?> CreateRegisteredSystem(RegisteredSystem toBeInserted, string[] defaultRights)
```

Comment:
i actually started fixing the issue with the registered system. So added required attribute on the RegisteredSystem. I think its better to take it as part of the PR. Add Required attribute in the RegisteredSystem